test/gurka/test_closure_annotations.cc

```diff
@@ -0,0 +1,327 @@
+#include "gurka.h"
+#include "test.h"
+
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+using LiveTrafficCustomize = test::LiveTrafficCustomize;
+
+namespace {
+
+inline void SetLiveSpeedFrom(baldr::TrafficSpeed* live_speed, uint64_t speed, uint8_t subsegment) {
```

Comment:
Any chance `subsegment` argument can be renamed to `breakpoint1` just to decrease number of terms describing the same thing?

Also, you're not setting the speed of the first subsegment here even though you do set the breakpoint1, is that intended? You may want to set `speed1` to either unknown or 0 (or something else) for clarity

Reply:
> Any chance `subsegment` argument can be renamed to `breakpoint1` just to decrease number of terms describing the same thing?
> 

Yeah, I can change `subsegment` to `breakpoint1` (this was just me trying to reverse engg the `TrafficSpeed` struct and trying to make sense of what breakpoint meant :stuck_out_tongue: ).

> Also, you're not setting the speed of the first subsegment here even though you do set the breakpoint1, is that intended? You may want to set `speed1` to either unknown or 0 (or something else) for clarity

So this function intends to set live speed to `0` **starting from** a particular subsegment/breakpoint of the edge. Anything before that should remain whatever value it previously had. TBH I arrived at the vlaues being set after a few iterations. Let me know if there's a better/correct/simpler way to do it. I want to partially close edges upto/from a certain percentage of the edge.